addons/controllers/csi/csiconfig_utils.go

```diff
@@ -0,0 +1,125 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/constants"
+	csiv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/csi/v1alpha1"
+)
+
+// mapCSIConfigToDataValues maps CSIConfig CR to data values
+func (r *CSIConfigReconciler) mapCSIConfigToDataValues(ctx context.Context, csiConfig *csiv1alpha1.CSIConfig) (*DataValues, error) {
```

Comment:
A overall question for all the map functions: Have we visited all the values and decided what are derived fields? Looks like all values come directly from config CR.

Reply:
updated